testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -6188,6 +6188,35 @@ void checkRegexpExtract(SqlOperatorFixture f0, FunctionAlias functionAlias) {
     f.checkNull("log(10, cast(null as real))");
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-6224">[CALCITE-6224]
+   * Add LOG@ function (enabled in MYSQL, Spark library)</a>. */
+  @Test void testLog2Func() {
+    final SqlOperatorFixture f0 = fixture();
+    final Consumer<SqlOperatorFixture> consumer = f -> {
+      f.setFor(SqlLibraryOperators.LOG2);
+      f.checkScalarApprox("log2(2)", "DOUBLE NOT NULL",
+          isWithin(1.0, 0.000001));
+      f.checkScalarApprox("log2(4)", "DOUBLE NOT NULL",
+          isWithin(2.0, 0.000001));
+      f.checkScalarApprox("log2(65536)", "DOUBLE NOT NULL",
+          isWithin(16.0, 0.000001));
+      f.checkScalarApprox("log2(-2)", "DOUBLE NOT NULL",
+          "NaN");
+      f.checkScalarApprox("log2(2/3)", "DOUBLE NOT NULL",
+          "-Infinity");
+      f.checkScalarApprox("log2(2.2)", "DOUBLE NOT NULL",
+          "1.1375035237499351");
+      f.checkScalarApprox("log2(0.5)", "DOUBLE NOT NULL",
+          "-1.0");
+      f.checkScalarApprox("log2(3)", "DOUBLE NOT NULL",
```

Comment:
You can see that there are retained decimal places. 
The first one is obviously log2(0) and log10(0). We may need to process this. 
Second, when converting fractions like 2/3 into sqlnode nodes, an integer value is taken, which also needs further modification.